utils/crypto/bls/signer/rpcsigner/bls_test.go

```diff
@@ -0,0 +1,201 @@
+	serverImpl, err := newServer()
+	require.NoError(err)
+
+	server := grpc.NewServer()
+
+	pb.RegisterSignerServer(server, serverImpl)
+	lis, err := net.Listen("tcp", ":0")
+	require.NoError(err)
+
+	go func() {
+		if err := server.Serve(lis); err != nil {
+			// Just log the error since we can't return it from the goroutine
+			// The error is not critical as it will happen during shutdown
+			_ = err
+		}
+	}()
+
+	conn, err := grpc.NewClient(lis.Addr().String(), grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		server.Stop()
+		require.NoError(err)
+		return nil
+	}
+
+	client, err := NewClient(conn)
+	if err != nil {
+		server.Stop()
+		require.NoError(err)
+		return nil
+	}
+
+	return &rpcSigner{
+		Client: client,
+		server: server,
+	}
+}
+
+func collectN[T any](n int, fn func() T) []T {
+	result := make([]T, n)
+	for i := 0; i < n; i++ {
+		result[i] = fn()
+	}
+	return result
+}
+
+func mapWithError[T any, U any](arr []T, fn func(T) (U, error)) ([]U, error) {
+	result := make([]U, len(arr))
+	for i, val := range arr {
+		newVal, err := fn(val)
+		if err != nil {
+			return nil, err
+		}
+		result[i] = newVal
+	}
+	return result, nil
+}
+
+func TestVerifyValidSignature(t *testing.T) {
+	require := require.New(t)
+	signer := NewRpcSigner(require)
+	defer signer.Close()
+
+	msg := []byte("TestVerifyValidSignature local signer")
+
+	sig, err := signer.Sign(msg)
+	require.NoError(err)
+
+	isValid := bls.Verify(signer.PublicKey(), sig, msg)
+	require.True(isValid)
+}
+
+func TestVerifyWrongMessageSignature(t *testing.T) {
+	require := require.New(t)
+	signer := NewRpcSigner(require)
+	defer signer.Close()
+
+	msg := []byte("TestVerifyWrongMessageSignature local signer")
+	wrongMsg := []byte("TestVerifyWrongMessageSignature local signer with wrong message")
+
+	sig, err := signer.Sign(msg)
+	require.NoError(err)
+
+	isValid := bls.Verify(signer.PublicKey(), sig, wrongMsg)
+	require.False(isValid)
+}
+
+func TestValidAggregation(t *testing.T) {
+	require := require.New(t)
+	keyPairs := collectN(2, func() bls.Signer {
+		return NewLocalSigner(require)
+	})
+
+	keyPairs = append(keyPairs, NewLocalSigner(require))
+
+	msg := []byte("TestValidAggregation local signer")
+
+	sigs, err := mapWithError(keyPairs, func(s bls.Signer) (*bls.Signature, error) {
+		return s.Sign(msg)
+	})
+	require.NoError(err)
+
+	pks, _ := mapWithError(keyPairs, func(s bls.Signer) (*bls.PublicKey, error) {
+		return s.PublicKey(), nil
+	})
+
+	isValid, err := AggregateAndVerify(pks, sigs, msg)
+	require.NoError(err)
+	require.True(isValid)
+}
+
+func TestSingleKeyAggregation(t *testing.T) {
+	require := require.New(t)
+	signer := NewRpcSigner(require)
+
+	pks := []*bls.PublicKey{signer.PublicKey()}
+
+	msg := []byte("TestSingleKeyAggregation local signer")
+
+	sig, err := signer.Sign(msg)
+	require.NoError(err)
+
+	isValid, err := AggregateAndVerify(pks, []*bls.Signature{sig}, msg)
+	require.NoError(err)
+	require.True(isValid)
+}
```

Comment:
Do we need tests on aggregation working? I feel like we already know aggregation will work because we are just implementing a type that produces `bls.Signature` and `bls.Publickey`, and those types + aggregation are already tested in the upstream package `bls`. I feel like coverage for `rpcsigner` only needs to cover that the `Sign*` functions produce signatures that are verifiable off of the `PublicKey`.